src/components/widget/MasterWidget.js

```diff
@@ -25,32 +26,52 @@ const dateParse = ['Date', 'DateTime', 'ZonedDateTime', 'Timestamp', 'Time'];
  * @module MasterWidget
  * @extends Component
  */
-class MasterWidget extends Component {
+class MasterWidget extends PureComponent {
   state = {
     updated: false,
     edited: false,
     data: '',
   };
 
   componentDidMount() {
-    const { data, widgetData, clearValue } = this.props;
+    const { data, clearValue } = this.props;
+    const localWidgetData = this.getWidgetData();
 
     // `clearValue` removes current field value for the widget. This is used when
     // user focuses on table cell and starts typing
     this.setState({
-      data: data || (clearValue ? '' : widgetData[0].value),
+      data: data || (clearValue ? '' : localWidgetData[0].value),
     });
   }
 
   UNSAFE_componentWillReceiveProps(nextProps) {
-    const { widgetData, widgetType } = this.props;
+    const { widgetType } = this.props;
+    const localWidgetData = this.getWidgetData();
     const { edited, data } = this.state;
-    let next = nextProps.widgetData[0].value;
+
+    const {
+      widgetData: nextWidgetData,
+      getWidgetData: nextGetWidgetData,
+      item: nextItem,
+      isEditable: nextIsEditable,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['isEditable' is missing in props validation (react/prop-types)](https://app.codacy.com/app/metasfresh/metasfresh-webui-frontend/pullRequest?prid=4689744)